float64 pn
float64 pe
float64 pd
float64 psi
float64 pn_dot
float64 pe_dot
float64 pd_dot
float64 psi_dot
float64 pn_ddot
float64 pe_ddot
float64 pd_ddot
float64 psi_ddot
